### PR TITLE
test: add tests to check rego custom key works as expected

### PR DESCRIPTION
**Pull request by TzlilSwimmer123**

### CODE REVIEW

File: pkg/jsonSchemaValidator/validator_test.go [hunk 1]
```diff
@@ -49,3 +56,76 @@ func TestValidateCustomKeysPass(t *testing.T) {
...
+		}
+		assert.Contains(t, err.Error(), "regoDefinition must be an object of type RegoDefinition json: cannot unmarshal number into Go struct field RegoDefinition.code of type string")
+	})
+	t.Run("validSchema", func(t *testing.T) {
+		c := jsonschema.NewCompiler()
+		c.RegisterExtension(extensions.RegoDefinitionCustomKey, extensions.CustomKeyRegoRule, extensions.CustomKeyRegoDefinitionCompiler{})
+		if err := c.AddResource("test.json", strings.NewReader(validRegoDefinitionJson)); err != nil {
+			t.Fatal(err)
+		}
+		sch, err := c.Compile("test.json")
```
Comment (royhadad) [inline pkg/jsonSchemaValidator/validator_test.go]: what is `sch`? #say_no_to_abbreviations

File: pkg/jsonSchemaValidator/validator_test.go [hunk 2]
```diff
@@ -49,3 +56,76 @@ func TestValidateCustomKeysPass(t *testing.T) {
...
+//go:embed test_fixtures/valid-rego-definition.json
+var validRegoDefinitionJson string
+
+//go:embed test_fixtures/rego-rule-fail.yaml
+var regoRuleFail string
+
+//go:embed test_fixtures/rego-rule-pass.yaml
+var regoRulePass string
+
+func TestRegoDefinitionCustomKey(t *testing.T) {
```
Comment (royhadad) [inline pkg/jsonSchemaValidator/validator_test.go:61]: I have an idea for improving those tests.
The public interface of this package, used by the rest of the code is as follows:
```
validationResult, err := e.jsonSchemaValidator.ValidateYamlSchema(string(ruleSchemaJson), string(configurationJson))
```
in essence: pass a rule's yaml schema, pass a file to test - get a result.

so it makes more sense to test the software the same way the users of the package use it (this time, the users are the pieces of code using this package)

File: pkg/jsonSchemaValidator/validator_test.go [hunk 3]
```diff
@@ -49,3 +134,67 @@ func TestValidateCustomKeysPass(t *testing.T) {
...
 }
+
+func TestValidateRegoDefinitionCustomKeyFail(t *testing.T) {
+	failResourceYamlFileContent, customRuleSchemaYamlFileContent, err :=
+		getResourceAndSchemaYamlContentsAsString(
+			regoYamlFilesPath+"/rego-rule-fail.yaml",
+			regoYamlFilesPath+"/valid-rego-definition.json",
+		)
+
+	if err != nil {
```
Comment (royhadad) [inline pkg/jsonSchemaValidator/validator_test.go]: this line can go inside `getResourceAndSchemaYamlContentsAsString `

**Review by royhadad — DISMISSED** [review]
maybe add some test cases for:
1. problems in the rego rule, `package` definition
2. json schema + rego (filter by kind)
3. multiple rego rules in one rule

**Review by royhadad — DISMISSED** [review]

**Review by shalev007 — APPROVED** [review]

**Pull request merged**